UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Webjob.CleanUpJob.UnitTests/Services/ExchangeSetCleanUpServiceTest.cs

```diff
@@ -0,0 +1,78 @@
+﻿using FakeItEasy;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using NUnit.Framework;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using UKHO.ExchangeSetService.CleanUpJob.Configuration;
+using UKHO.ExchangeSetService.CleanUpJob.Helpers;
+using UKHO.ExchangeSetService.CleanUpJob.Services;
+using UKHO.ExchangeSetService.Common.Configuration;
+using UKHO.ExchangeSetService.Common.Storage;
+
+namespace UKHO.ExchangeSetService.Webjob.CleanUpJob.UnitTests.Services
+{
+    public class ExchangeSetCleanUpServiceTest
+    {
+        public IAzureFileSystemHelper fakeAzureFileSystemHelper;
+        public IOptions<EssFulfilmentStorageConfiguration> fakeStorageConfig;
+        public ISalesCatalogueStorageService fakeScsStorageService;
+        public IConfiguration fakeConfiguration;
+        public ILogger<ExchangeSetCleanUpService> fakeLogger;
+        public ExchangeSetCleanUpService exchangeSetCleanUpService;
+        public IOptions<CleanUpConfiguration> fakeCleanUpConfig;
+        public string fakeFilePath = @"D:\\Downloads";
+        public string fakeStorageAccountConnectionString = "DefaultEndpointsProtocol = https; AccountName = testessdevstorage2; AccountKey =testaccountkey; EndpointSuffix = core.windows.net";
+
+        [SetUp]
+        public void Setup()
+        {
+            fakeAzureFileSystemHelper = A.Fake<IAzureFileSystemHelper>();
+            fakeStorageConfig = Options.Create(new EssFulfilmentStorageConfiguration()
+                                         { StorageContainerName ="Test"  });
+            fakeScsStorageService = A.Fake<ISalesCatalogueStorageService>();
+            fakeConfiguration = A.Fake<IConfiguration>();
+            fakeLogger = A.Fake<ILogger<ExchangeSetCleanUpService>>();
+            fakeCleanUpConfig = Options.Create(new CleanUpConfiguration()
+            { NumberOfDays = 1 });
+
+            exchangeSetCleanUpService = new ExchangeSetCleanUpService(fakeAzureFileSystemHelper, fakeStorageConfig, fakeScsStorageService, fakeConfiguration, fakeLogger, fakeCleanUpConfig);
+        }
+
+        [Test]
+        public void WhenScsStorageAccountAccessKeyValueNotFound_ThenReturnKeyNotFoundException()
+        {
+            A.CallTo(() => fakeScsStorageService.GetStorageAccountConnectionString(A<string>.Ignored, A<string>.Ignored))
+              .Throws(new KeyNotFoundException("Storage account accesskey not found"));
+
+            Assert.ThrowsAsync(Is.TypeOf<KeyNotFoundException>()
+                   .And.Message.EqualTo("Storage account accesskey not found")
+                    , async delegate { await exchangeSetCleanUpService.DeleteHistoricFoldersAndFiles(); });
+        }
```

Comment:
Is this test not testing the AzureFileSystemHelper - see comments made earlier on the AzureFileSystemhelper method!

Reply:
Resolved, As discussed we have added FakeAzureFileHelper for checking methods which actually calls and returns bool value from that.